keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c

```diff
@@ -0,0 +1,177 @@
+#include QMK_KEYBOARD_H
+
+void tap_key(uint16_t keycode) {
+  register_code  (keycode);
+  unregister_code(keycode);
+}
+
+void shift_key(uint16_t keycode) {
+  register_code  (KC_LSFT);
+  tap_key        (keycode);
+  unregister_code(KC_LSFT);
+}
+
+
+// Layer declarations
+enum {
+    DEF_LAYER,
+    MOD_LAYER,
+};
+
+// Tap Dance declarations
+enum {
+    TD_CAR,
+    TD_PRINT,
+    TD_CAM_UP,
+    TD_CAM_DN
+};
+
+enum custom_keycodes {
+    CHASE = 0,
+    REVERSE
+};
+
+void cam_up(qk_tap_dance_state_t *state, void *user_data) {
+    switch (state->count) {
+        case 1:
+            tap_key(KC_C);           // next cam
+            break;
+        case 2:
+            shift_key(KC_8);
```

Comment:
https://docs.qmk.fm/#/keycodes_us_ansi_shifted?id=keycodes

Reply:
This change looks straight forward enough but it doesn't compile:

## Diff
```
diff --git a/keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c b/keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c
index 983d73a3f..0e99eb68e 100644
--- a/keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c
+++ b/keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c
@@ -46,9 +46,9 @@ void cam_up(qk_tap_dance_state_t *state, void *user_data) {
             tap_code(KC_C);           // next cam
             break;
         case 2:
-            tap_code16(LSFT(KC_8));
+            tap_code(KC_ASTR);
             wait_ms(CAM_DELAY);
-            tap_code16(LSFT(KC_8));
+            tap_code(KC_ASTR);
             wait_ms(CAM_DELAY);
             tap_code(KC_2);
             wait_ms(CAM_DELAY);
@@ -65,9 +65,9 @@ void cam_down(qk_tap_dance_state_t *state, void *user_data) {
             tap_code16(LSFT(KC_C));           // next cam
             break;
         case 2:
-            tap_code16(LSFT(KC_8));
+            tap_code(KC_ASTR);
             wait_ms(CAM_DELAY);
-            tap_code16(LSFT(KC_8));
+            tap_code(KC_ASTR);
             wait_ms(CAM_DELAY);
             tap_code(KC_2);
             wait_ms(CAM_DELAY);
```
## Error
```
Ψ Compiling keymap with make -j 1 kbdfans/kbdpad_mk2:iracing_replay


QMK Firmware 0.12.29
WARNING: Some git submodules are out of date or modified.
 Please consider running make git-submodule.

Making kbdfans/kbdpad_mk2 with keymap iracing_replay

avr-gcc (GCC) 10.2.0
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Size before:
   text	  data	   bss	   dec	   hex	filename
      0	 26564	     0	 26564	  67c4	.build/kbdfans_kbdpad_mk2_iracing_replay.hex

Compiling: keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c                            In file included from quantum/keymap.h:47,
                 from quantum/quantum.h:30,
                 from keyboards/kbdfans/kbdpad_mk2/kbdpad_mk2.h:18,
                 from keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c:17:
keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c: In function 'cam_up':
quantum/quantum_keycodes.h:582:18: error: unsigned conversion from 'int' to 'uint8_t' {aka 'unsigned char'} changes value from '549' to '37' [-Werror=overflow]
  582 | #define LSFT(kc) (QK_LSFT | (kc))
      |                  ^~~~~~~~~~~~~~~~
quantum/quantum_keycodes.h:639:17: note: in expansion of macro 'LSFT'
  639 | #define KC_ASTR LSFT(KC_8)  // *
      |                 ^~~~
keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c:49:22: note: in expansion of macro 'KC_ASTR'
   49 |             tap_code(KC_ASTR);
      |                      ^~~~~~~
quantum/quantum_keycodes.h:582:18: error: unsigned conversion from 'int' to 'uint8_t' {aka 'unsigned char'} changes value from '549' to '37' [-Werror=overflow]
  582 | #define LSFT(kc) (QK_LSFT | (kc))
      |                  ^~~~~~~~~~~~~~~~
quantum/quantum_keycodes.h:639:17: note: in expansion of macro 'LSFT'
  639 | #define KC_ASTR LSFT(KC_8)  // *
      |                 ^~~~
keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c:51:22: note: in expansion of macro 'KC_ASTR'
   51 |             tap_code(KC_ASTR);
      |                      ^~~~~~~
keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c: In function 'cam_down':
quantum/quantum_keycodes.h:582:18: error: unsigned conversion from 'int' to 'uint8_t' {aka 'unsigned char'} changes value from '549' to '37' [-Werror=overflow]
  582 | #define LSFT(kc) (QK_LSFT | (kc))
      |                  ^~~~~~~~~~~~~~~~
quantum/quantum_keycodes.h:639:17: note: in expansion of macro 'LSFT'
  639 | #define KC_ASTR LSFT(KC_8)  // *
      |                 ^~~~
keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c:68:22: note: in expansion of macro 'KC_ASTR'
   68 |             tap_code(KC_ASTR);
      |                      ^~~~~~~
quantum/quantum_keycodes.h:582:18: error: unsigned conversion from 'int' to 'uint8_t' {aka 'unsigned char'} changes value from '549' to '37' [-Werror=overflow]
  582 | #define LSFT(kc) (QK_LSFT | (kc))
      |                  ^~~~~~~~~~~~~~~~
quantum/quantum_keycodes.h:639:17: note: in expansion of macro 'LSFT'
  639 | #define KC_ASTR LSFT(KC_8)  // *
      |                 ^~~~
keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.c:70:22: note: in expansion of macro 'KC_ASTR'
   70 |             tap_code(KC_ASTR);
      |                      ^~~~~~~
cc1: all warnings being treated as errors
 [ERRORS]
 | 
 | 
 | 
make[1]: *** [tmk_core/rules.mk:417: .build/obj_kbdfans_kbdpad_mk2_iracing_replay/keyboards/kbdfans/kbdpad_mk2/keymaps/iracing_replay/keymap.o] Error 1
Make finished with errors
make: *** [Makefile:523: kbdfans/kbdpad_mk2:iracing_replay] Error 1
```

Reply:
Nevermind, figured it out.  Needed to stick with `tap_code16()`.